core/geometry/src/geometry3d/OrderedRotationAngles.ts

```diff
@@ -119,139 +180,144 @@ export class OrderedRotationAngles {
 
     switch (order) {
       case AxisOrder.XYZ: {
-        yRad = Math.asin(Math.max(-1, Math.min(1, m13)));
+        yRad = Math.asin(Math.max(-1, Math.min(1, -m31))); // limit asin domain to [-1,1]
 
-        if (Math.abs(m13) < 0.99999) {
-          xRad = Math.atan2(- m23, m33);
-          zRad = Math.atan2(- m12, m11);
+        if (Math.abs(m31) < 0.99999) {
+          xRad = Math.atan2(m32, m33);
+          zRad = Math.atan2(m21, m11);
         } else {
-          xRad = Math.atan2(m32, m22);
+          /*
+          * If Math.abs(m31) = 1 then yRad = 90 degrees and therefore, we have a gimbal lock.
```

Comment:
yRad could also be 270 deg.  (Resulting in same gimbal lock situation)

Reply:
correct. I make the comment yRad=+-90 in the next PR